katcp/test/test_resource_client.py

```diff
@@ -73,18 +73,22 @@ class test_KATCPClientResourceRequest(unittest.TestCase):
     def setUp(self):
         self.mock_client = mock.Mock()
         self.DUT = resource_client.KATCPClientResourceRequest(
-            'the-request', 'The description', self.mock_client)
+            {'name': 'the-request',
+             'description': 'The description',
+             'timeout_hint': 33.34},
+            self.mock_client)
 
     def test_init(self):
         self.assertEqual(self.DUT.name, 'the-request')
         self.assertEqual(self.DUT.description, 'The description')
+        self.assertEqual(self.DUT.timeout_hint, 33.34)
         # Check that we are registered to the correct ABC
         self.assertIsInstance(self.DUT, resource.KATCPRequest)
 
     def test_request(self):
         reply = self.DUT('parm1', 2)
         self.mock_client.wrapped_request.assert_called_once_with(
-            'the-request', 'parm1', 2)
+            'the-request', 'parm1', 2, timeout=33.34)
         self.assertIs(reply, self.mock_client.wrapped_request.return_value)
 
```

Comment:
Would it be useful to add another test or test class like `test_KATCPClientResourceRequest`, that uses a timeout hint of `None`, just to check the timeout parameter comes through as expected?